stdlib/public/core/Sequence.swift

```diff
@@ -1194,10 +1184,7 @@ extension Sequence where Element : Equatable {
   }
 }
 
-extension Sequence where
-  SubSequence : Sequence,
-  SubSequence.Element == Element,
-  SubSequence.SubSequence == SubSequence {
+extension Sequence where SubSequence == AnySequence<Element> {
```

Comment:
Re that last one, I suppose it does, since it means you couldn't just use that method in a `Sequence where Element: Equatable` extension/generic. Hmmm. There's no way to implement it just in terms of returning `SubSequence`, since it needs to eagerly consume the sequence in order to allow random access to the chunks in the array. Maybe that should be converted to return `[[Element]]` instead? Not sure what we're getting out of it returning an `AnySequence` wrapping an array if it doesn't need to match a `SubSequence` requirement anyway.

Reply:
> Maybe that should be converted to return `[[Element]]` instead? 

Hmmm. There's possible a good case for the default split on sequences being lazy and non-consuming. Anyway, not such a slam-dunk as the `prefix`/`suffix` cases.

Reply:
It has to consume the sequence, since it returns an array and someone might write something like:

```swift
let result = mySinglePassSequence.split(separator: " ")
let lastChunk = result.last!
```

We could have a lazy version that returns a non-array sequence, but these ones need to be eager.